packages/edit-widgets/src/components/keyboard-shortcuts/index.js

```diff
@@ -1,23 +1,55 @@
+/**
+ * External dependencies
+ */
+import { first, last } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
 /**
  * WordPress dependencies
  */
-import { useEffect } from '@wordpress/element';
+import { useEffect, useCallback } from '@wordpress/element';
 import {
 	useShortcut,
 	store as keyboardShortcutsStore,
 } from '@wordpress/keyboard-shortcuts';
-import { useDispatch } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { store as blockEditorStore } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { store as editWidgetsStore } from '../../store';
 
+let selectedClientId;
```

Comment:
A better "React" way of doing this is to convert this to a ref with `useRef`. I'm not sure why we need this variable either? Couldn't we just use `newSelectedClientId`?

Reply:
We can't because it's updated on render once the selection is changed. We need to maintain a reference to the "selected" block _when the keyboard shortcut was fired_.

Reply:
Hmm, I'm not sure I follow.

Do you mean we can't convert this to a ref or we can't use `newSelectedClientId`? I think either way this deserves some comments 😆.